asyncio_mqtt/client.py

```diff
@@ -230,31 +238,32 @@ def __init__(
 
         if tls_context is not None:
             self._client.tls_set_context(tls_context)
-        
+
         if tls_params is not None:
-            self._client.tls_set(ca_certs= tls_params.ca_certs,
-                                 certfile= tls_params.certfile,
-                                 keyfile= tls_params.keyfile,
-                                 cert_reqs= tls_params.cert_reqs,
-                                 tls_version= tls_params.tls_version,
-                                 ciphers= tls_params.ciphers,
-                                 keyfile_password= tls_params.keyfile_password)
-        
+            self._client.tls_set(
+                ca_certs=tls_params.ca_certs,
+                certfile=tls_params.certfile,
+                keyfile=tls_params.keyfile,
+                cert_reqs=tls_params.cert_reqs,
+                tls_version=tls_params.tls_version,
+                ciphers=tls_params.ciphers,
+                keyfile_password=tls_params.keyfile_password,
+            )
+
         if proxy is not None:
-            self._client.proxy_set(**proxy.proxy_args)    
+            self._client.proxy_set(**proxy.proxy_args)
 
-        if websocket_path is not None or websocket_headers is not None:
-            self._client.ws_set_options(
-                path=websocket_path, headers=websocket_headers
-            )
+        if websocket_path is not None:
+            self._client.ws_set_options(path=websocket_path, headers=websocket_headers)
 
         if will is not None:
             self._client.will_set(
                 will.topic, will.payload, will.qos, will.retain, will.properties
             )
 
         self._client.message_retry_set(message_retry_set)
-        self._socket_options: Tuple[SocketOption] = tuple(socket_options)
+        if socket_options is not None:
+            self._socket_options: Tuple[SocketOption, ...] = tuple(socket_options)
 
```

Comment:
Instead, change the default of `socket_options` to `None` (in the function parameters).
Then, use:
```python
if socket_options is not None:
    socket_options = tuple()
self._socket_options: Tuple[SocketOption, ...] = tuple(socket_options)
```

Reply:
Note that way *always* assign `self._socket_options` now. This is important. :)

Reply:
Do you mean
```python
if socket_options is None:
    socket_options = tuple()
self._socket_options: Tuple[SocketOption, ...] = tuple(socket_options)
```

Reply:
Yes. 😁 Good catch. 👍